BootloaderCommonPkg/Library/ContainerLib/ContainerLib.c

```diff
@@ -490,6 +490,84 @@ RegisterContainer (
   return Status;
 }
 
+/**
+  This function would check container validity.
+
+  @param[in]  Signature   Container signature
+
+  @retval  BOOLEAN        Container valid/not valid
+
+**/
+BOOLEAN
+IsContainerValid (
+  IN  UINT32   Signature
+  )
+{
+  if((Signature == CONTAINER_BOOT_SIGNATURE)
+            || (Signature == CONTAINER_MONO_SIGN_SIGNATURE)
+            || (Signature == CONTAINER_KEY_HASH_STORE_SIGNATURE)
+            || ((Signature & 0x00FFFFFF) == CONTAINER_OEM_BASE_SIGNATURE)
+            || (Signature ==CONTAINER_IPFW_SIGNATURE))
+  {
+    return TRUE;
+  } else {
+    return FALSE;
+  }
+}
+
+/**
+  Locate a container entry information based on signature
+
+  @param[in]     ContainerSig       Container signature.
+  @param[in,out] ContainerEntryPtr  Pointer to receive container entry info.
+
+  @retval EFI_UNSUPPORTED          Unsupported container.
+  @retval EFI_NOT_FOUND            Cannot locate container.
+  @retval EFI_SECURITY_VIOLATION   Authentication failed.
+  @retval EFI_SUCCESS              component region is located successfully.
+
+**/
+EFI_STATUS
+LocateContainerEntry (

```

Comment:
You don't  need this function.  The following will give you the container header.
LocateComponentEntry (ContainerSig, 0, NULL, NULL);
ContainerEntry = GetContainerBySignature (ContainerSig);

Reply:
Thanks. I would use as suggested.

Reply:
Updated LocateComponentEntry to handle when component=0